2025-05-28 09:10:24,275 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/token
2025-05-28 09:10:24,277 - fire_safety - INFO - Response: 200
2025-05-28 09:10:24,279 - fire_safety - INFO - Request: POST http://localhost:8000/token
2025-05-28 09:10:25,184 - fire_safety - INFO - Response: 200
2025-05-28 09:10:36,715 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 09:10:36,718 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 09:10:36,718 - fire_safety - INFO - Response: 200
2025-05-28 09:10:36,720 - fire_safety - INFO - Response: 200
2025-05-28 09:10:36,724 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:10:36,725 - fire_safety - INFO - Response: 307
2025-05-28 09:10:36,729 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:10:36,730 - fire_safety - INFO - Response: 307
2025-05-28 09:10:36,732 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 09:10:36,733 - fire_safety - INFO - Response: 200
2025-05-28 09:10:36,763 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:10:36,859 - fire_safety - INFO - Response: 200
2025-05-28 09:10:36,862 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:10:36,867 - fire_safety - INFO - Response: 200
2025-05-28 09:10:39,092 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 09:10:39,093 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 09:10:39,093 - fire_safety - INFO - Response: 200
2025-05-28 09:10:39,095 - fire_safety - INFO - Response: 200
2025-05-28 09:10:39,096 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:10:39,104 - fire_safety - INFO - Response: 200
2025-05-28 09:10:39,124 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:10:39,129 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 09:10:39,131 - fire_safety - INFO - Response: 200
2025-05-28 09:10:39,133 - fire_safety - INFO - Response: 200
2025-05-28 09:10:39,135 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:10:39,135 - fire_safety - INFO - Response: 307
2025-05-28 09:10:39,145 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 09:10:39,169 - fire_safety - INFO - Response: 200
2025-05-28 09:10:39,170 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:10:39,171 - fire_safety - INFO - Response: 307
2025-05-28 09:10:39,172 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:10:39,221 - fire_safety - INFO - Response: 200
2025-05-28 09:10:39,224 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:10:39,230 - fire_safety - INFO - Response: 200
2025-05-28 09:10:56,818 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors
2025-05-28 09:10:56,819 - fire_safety - INFO - Response: 200
2025-05-28 09:10:56,821 - fire_safety - INFO - Request: POST http://localhost:8000/sensors
2025-05-28 09:10:56,822 - fire_safety - INFO - Response: 307
2025-05-28 09:10:56,828 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/
2025-05-28 09:10:56,829 - fire_safety - INFO - Response: 200
2025-05-28 09:10:56,831 - fire_safety - INFO - Request: POST http://localhost:8000/sensors/
2025-05-28 09:10:56,836 - fire_safety - INFO - Response: 422
2025-05-28 09:13:40,500 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:13:40,501 - fire_safety - INFO - Response: 307
2025-05-28 09:13:40,503 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:13:40,504 - fire_safety - INFO - Response: 307
2025-05-28 09:13:40,509 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:13:40,516 - fire_safety - INFO - Response: 200
2025-05-28 09:13:40,520 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:13:40,526 - fire_safety - INFO - Response: 200
2025-05-28 09:13:42,159 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:13:42,166 - fire_safety - INFO - Response: 200
2025-05-28 09:13:42,171 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:13:42,177 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:13:42,178 - fire_safety - INFO - Response: 307
2025-05-28 09:13:42,180 - fire_safety - INFO - Response: 200
2025-05-28 09:13:42,185 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:13:42,189 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:13:42,190 - fire_safety - INFO - Response: 307
2025-05-28 09:13:42,192 - fire_safety - INFO - Response: 200
2025-05-28 09:13:42,222 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:13:42,228 - fire_safety - INFO - Response: 200
2025-05-28 09:13:45,539 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:13:45,540 - fire_safety - INFO - Response: 307
2025-05-28 09:13:45,543 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:13:45,544 - fire_safety - INFO - Response: 307
2025-05-28 09:13:45,569 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:13:45,575 - fire_safety - INFO - Response: 200
2025-05-28 09:13:45,578 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:13:45,584 - fire_safety - INFO - Response: 200
2025-05-28 09:13:58,801 - fire_safety - INFO - Request: GET http://127.0.0.1:8000/buildings/?skip=0&limit=100
2025-05-28 09:13:58,827 - fire_safety - INFO - Response: 200
2025-05-28 09:14:17,955 - fire_safety - INFO - Request: GET http://127.0.0.1:8000/buildings/7
2025-05-28 09:14:17,973 - fire_safety - INFO - Response: 200
2025-05-28 09:14:57,174 - fire_safety - INFO - Request: PUT http://127.0.0.1:8000/buildings/7
2025-05-28 09:14:57,185 - fire_safety - ERROR - Request failed: 'BuildingCreate' object has no attribute 'owner_id'
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 308, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 301, in app
    raw_response = await run_endpoint_function(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 214, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\concurrency.py", line 37, in run_in_threadpool
    return await anyio.to_thread.run_sync(func)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\_backends\_asyncio.py", line 2470, in run_sync_in_worker_thread
    return await future
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\_backends\_asyncio.py", line 967, in run
    result = context.run(func, *args)
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 109, in update_building
    db_building = crud.update_building(db, building_id=building_id, building=building)
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\crud.py", line 77, in update_building
    db_building.owner_id = building.owner_id
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\pydantic\main.py", line 989, in __getattr__
    raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
AttributeError: 'BuildingCreate' object has no attribute 'owner_id'
2025-05-28 09:14:57,230 - fire_safety - ERROR - Unhandled exception: 'BuildingCreate' object has no attribute 'owner_id'
NoneType: None
2025-05-28 09:15:32,510 - fire_safety - INFO - Request: POST http://127.0.0.1:8000/sensors/
2025-05-28 09:15:32,904 - fire_safety - INFO - Response: 201
2025-05-28 09:15:43,108 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:15:43,114 - fire_safety - INFO - Response: 200
2025-05-28 09:15:43,117 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:15:43,123 - fire_safety - INFO - Response: 200
2025-05-28 09:15:43,127 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:15:43,128 - fire_safety - INFO - Response: 307
2025-05-28 09:15:43,133 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:15:43,134 - fire_safety - INFO - Response: 307
2025-05-28 09:15:43,137 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:15:43,143 - fire_safety - INFO - Response: 200
2025-05-28 09:15:43,146 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:15:43,152 - fire_safety - INFO - Response: 200
2025-05-28 09:15:47,085 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:15:47,086 - fire_safety - INFO - Response: 307
2025-05-28 09:15:47,091 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:15:47,092 - fire_safety - INFO - Response: 307
2025-05-28 09:15:47,098 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:15:47,108 - fire_safety - INFO - Response: 200
2025-05-28 09:15:47,110 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:15:47,117 - fire_safety - INFO - Response: 200
2025-05-28 09:22:33,183 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 09:22:33,184 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 09:22:33,185 - fire_safety - INFO - Response: 200
2025-05-28 09:22:33,186 - fire_safety - INFO - Response: 200
2025-05-28 09:22:33,188 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:22:33,189 - fire_safety - INFO - Response: 307
2025-05-28 09:22:33,193 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:22:33,194 - fire_safety - INFO - Response: 307
2025-05-28 09:22:33,196 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 09:22:33,199 - fire_safety - INFO - Response: 200
2025-05-28 09:22:33,202 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 09:22:33,203 - fire_safety - INFO - Response: 200
2025-05-28 09:22:33,205 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:22:33,213 - fire_safety - INFO - Response: 200
2025-05-28 09:22:33,216 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:22:33,222 - fire_safety - INFO - Response: 200
2025-05-28 09:22:41,969 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/token
2025-05-28 09:22:42,114 - fire_safety - INFO - Response: 200
2025-05-28 09:22:42,260 - fire_safety - INFO - Request: POST http://localhost:8000/token
2025-05-28 09:22:42,748 - fire_safety - INFO - Response: 200
2025-05-28 09:22:44,052 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:22:44,053 - fire_safety - INFO - Response: 307
2025-05-28 09:22:44,058 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 09:22:44,059 - fire_safety - INFO - Response: 307
2025-05-28 09:22:44,062 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:22:44,076 - fire_safety - INFO - Response: 200
2025-05-28 09:22:44,086 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 09:22:44,092 - fire_safety - INFO - Response: 200
2025-05-28 09:22:45,609 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 09:22:45,610 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 09:22:45,610 - fire_safety - INFO - Response: 200
2025-05-28 09:22:45,612 - fire_safety - INFO - Response: 200
2025-05-28 09:22:45,614 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:22:45,628 - fire_safety - INFO - Response: 200
2025-05-28 09:22:45,676 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:22:45,680 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 09:22:45,681 - fire_safety - INFO - Response: 200
2025-05-28 09:22:45,684 - fire_safety - INFO - Response: 200
2025-05-28 09:22:45,706 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:22:45,707 - fire_safety - INFO - Response: 307
2025-05-28 09:22:45,711 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:22:45,712 - fire_safety - INFO - Response: 307
2025-05-28 09:22:45,714 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 09:22:45,715 - fire_safety - INFO - Response: 200
2025-05-28 09:22:45,717 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:22:45,723 - fire_safety - INFO - Response: 200
2025-05-28 09:22:45,726 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:22:45,731 - fire_safety - INFO - Response: 200
2025-05-28 09:23:06,428 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/
2025-05-28 09:23:06,429 - fire_safety - INFO - Response: 200
2025-05-28 09:23:06,431 - fire_safety - INFO - Request: POST http://localhost:8000/sensors/
2025-05-28 09:23:06,442 - fire_safety - INFO - Response: 201
2025-05-28 09:23:06,473 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:23:06,487 - fire_safety - INFO - Response: 200
2025-05-28 09:23:06,507 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 09:23:06,514 - fire_safety - INFO - Response: 200
2025-05-28 09:23:06,516 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:23:06,517 - fire_safety - INFO - Response: 307
2025-05-28 09:23:06,552 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 09:23:06,553 - fire_safety - INFO - Response: 307
2025-05-28 09:23:06,561 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:23:06,566 - fire_safety - INFO - Response: 200
2025-05-28 09:23:06,568 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 09:23:06,586 - fire_safety - INFO - Response: 200
2025-05-28 19:01:55,865 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 19:01:55,867 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 19:01:55,868 - fire_safety - INFO - Response: 200
2025-05-28 19:01:55,869 - fire_safety - INFO - Response: 200
2025-05-28 19:01:55,871 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:01:55,872 - fire_safety - INFO - Response: 307
2025-05-28 19:01:55,882 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:01:55,883 - fire_safety - INFO - Response: 307
2025-05-28 19:01:55,885 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 19:01:55,886 - fire_safety - INFO - Response: 200
2025-05-28 19:01:55,892 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 19:01:55,893 - fire_safety - INFO - Response: 200
2025-05-28 19:01:55,897 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:01:56,325 - fire_safety - INFO - Response: 200
2025-05-28 19:01:56,331 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:01:56,338 - fire_safety - INFO - Response: 200
2025-05-28 19:01:58,674 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:01:58,675 - fire_safety - INFO - Response: 307
2025-05-28 19:01:58,679 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:01:58,681 - fire_safety - INFO - Response: 307
2025-05-28 19:01:58,685 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:01:58,697 - fire_safety - INFO - Response: 200
2025-05-28 19:01:58,699 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:01:58,707 - fire_safety - INFO - Response: 200
2025-05-28 19:01:59,895 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 19:01:59,896 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 19:01:59,897 - fire_safety - INFO - Response: 200
2025-05-28 19:01:59,898 - fire_safety - INFO - Response: 200
2025-05-28 19:01:59,909 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:01:59,920 - fire_safety - INFO - Response: 200
2025-05-28 19:01:59,928 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:01:59,933 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 19:01:59,935 - fire_safety - INFO - Response: 200
2025-05-28 19:01:59,938 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:01:59,939 - fire_safety - INFO - Response: 307
2025-05-28 19:01:59,939 - fire_safety - INFO - Response: 200
2025-05-28 19:01:59,946 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 19:01:59,947 - fire_safety - INFO - Response: 200
2025-05-28 19:01:59,950 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:01:59,951 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:01:59,951 - fire_safety - INFO - Response: 307
2025-05-28 19:02:00,047 - fire_safety - INFO - Response: 200
2025-05-28 19:02:00,048 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:02:00,056 - fire_safety - INFO - Response: 200
2025-05-28 19:02:01,965 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:02:01,978 - fire_safety - INFO - Response: 200
2025-05-28 19:02:01,988 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:02:01,994 - fire_safety - INFO - Response: 200
2025-05-28 19:02:02,000 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:02:02,001 - fire_safety - INFO - Response: 307
2025-05-28 19:02:02,007 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:02:02,008 - fire_safety - INFO - Response: 307
2025-05-28 19:02:02,013 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:02:02,019 - fire_safety - INFO - Response: 200
2025-05-28 19:02:02,021 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:02:02,029 - fire_safety - INFO - Response: 200
2025-05-28 19:02:03,026 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:02:03,035 - fire_safety - INFO - Response: 200
2025-05-28 19:02:03,039 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:02:03,047 - fire_safety - INFO - Response: 200
2025-05-28 19:02:03,048 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:02:03,049 - fire_safety - INFO - Response: 307
2025-05-28 19:02:03,057 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:02:03,058 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:02:03,058 - fire_safety - INFO - Response: 307
2025-05-28 19:02:03,064 - fire_safety - INFO - Response: 200
2025-05-28 19:02:03,069 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:02:03,074 - fire_safety - INFO - Response: 200
2025-05-28 19:09:16,937 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/token
2025-05-28 19:09:16,938 - fire_safety - INFO - Response: 200
2025-05-28 19:09:16,941 - fire_safety - INFO - Request: POST http://localhost:8000/token
2025-05-28 19:09:17,528 - fire_safety - INFO - Response: 200
2025-05-28 19:09:20,199 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:09:20,200 - fire_safety - INFO - Response: 307
2025-05-28 19:09:20,204 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:09:20,206 - fire_safety - INFO - Response: 307
2025-05-28 19:09:20,212 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:09:20,223 - fire_safety - INFO - Response: 200
2025-05-28 19:09:20,224 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:09:20,231 - fire_safety - INFO - Response: 200
2025-05-28 19:09:22,998 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:09:23,009 - fire_safety - INFO - Response: 200
2025-05-28 19:09:23,013 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:09:23,021 - fire_safety - INFO - Response: 200
2025-05-28 19:09:23,023 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:09:23,024 - fire_safety - INFO - Response: 307
2025-05-28 19:09:23,030 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:09:23,031 - fire_safety - INFO - Response: 307
2025-05-28 19:09:23,033 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:09:23,040 - fire_safety - INFO - Response: 200
2025-05-28 19:09:23,046 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:09:23,051 - fire_safety - INFO - Response: 200
2025-05-28 19:09:29,245 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:09:29,256 - fire_safety - INFO - Response: 200
2025-05-28 19:09:29,266 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:09:29,273 - fire_safety - INFO - Response: 200
2025-05-28 19:09:29,275 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:09:29,276 - fire_safety - INFO - Response: 307
2025-05-28 19:09:29,279 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:09:29,280 - fire_safety - INFO - Response: 307
2025-05-28 19:09:29,282 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:09:29,289 - fire_safety - INFO - Response: 200
2025-05-28 19:09:29,291 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:09:29,296 - fire_safety - INFO - Response: 200
2025-05-28 19:09:29,882 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:09:29,885 - fire_safety - INFO - Response: 307
2025-05-28 19:09:29,889 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:09:29,890 - fire_safety - INFO - Response: 307
2025-05-28 19:09:29,892 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:09:29,901 - fire_safety - INFO - Response: 200
2025-05-28 19:09:29,904 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:09:29,909 - fire_safety - INFO - Response: 200
2025-05-28 19:34:39,158 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 19:34:39,160 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 19:34:39,161 - fire_safety - INFO - Response: 200
2025-05-28 19:34:39,163 - fire_safety - INFO - Response: 200
2025-05-28 19:34:39,165 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:34:39,179 - fire_safety - INFO - Response: 422
2025-05-28 19:34:39,182 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:34:39,185 - fire_safety - INFO - Response: 422
2025-05-28 19:34:40,883 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 19:34:40,884 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 19:34:40,885 - fire_safety - INFO - Response: 200
2025-05-28 19:34:40,886 - fire_safety - INFO - Response: 200
2025-05-28 19:34:40,887 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:34:40,888 - fire_safety - INFO - Response: 307
2025-05-28 19:34:40,890 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:34:40,890 - fire_safety - INFO - Response: 307
2025-05-28 19:34:40,896 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 19:34:40,897 - fire_safety - INFO - Response: 200
2025-05-28 19:34:40,898 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 19:34:40,900 - fire_safety - INFO - Response: 200
2025-05-28 19:34:40,903 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:34:40,931 - fire_safety - INFO - Response: 200
2025-05-28 19:34:40,934 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:34:40,942 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,504 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 19:34:42,505 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:42,505 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,507 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 19:34:42,508 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,511 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,512 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:42,514 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,517 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:42,518 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:42,519 - fire_safety - INFO - Response: 307
2025-05-28 19:34:42,526 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:42,530 - fire_safety - INFO - Response: 307
2025-05-28 19:34:42,532 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:42,533 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,539 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,541 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:42,542 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:42,557 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,559 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:42,566 - fire_safety - INFO - Response: 200
2025-05-28 19:34:42,642 - fire_safety - INFO - Response: 200
2025-05-28 19:34:44,946 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/1
2025-05-28 19:34:44,947 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/1
2025-05-28 19:34:44,948 - fire_safety - INFO - Response: 200
2025-05-28 19:34:44,951 - fire_safety - INFO - Response: 200
2025-05-28 19:34:44,953 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:34:44,970 - fire_safety - INFO - Response: 200
2025-05-28 19:34:44,973 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:34:44,981 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:34:44,982 - fire_safety - INFO - Response: 200
2025-05-28 19:34:44,985 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:34:44,986 - fire_safety - INFO - Response: 307
2025-05-28 19:34:44,989 - fire_safety - INFO - Response: 200
2025-05-28 19:34:44,993 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:34:44,994 - fire_safety - INFO - Response: 200
2025-05-28 19:34:44,996 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:34:44,998 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:34:45,000 - fire_safety - INFO - Response: 307
2025-05-28 19:34:45,036 - fire_safety - INFO - Response: 200
2025-05-28 19:34:45,037 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:34:45,044 - fire_safety - INFO - Response: 200
2025-05-28 19:34:47,062 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:47,063 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:47,064 - fire_safety - INFO - Response: 307
2025-05-28 19:34:47,068 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:47,072 - fire_safety - INFO - Response: 307
2025-05-28 19:34:47,074 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:47,079 - fire_safety - INFO - Response: 200
2025-05-28 19:34:47,086 - fire_safety - INFO - Response: 200
2025-05-28 19:34:47,088 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:47,089 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:47,100 - fire_safety - INFO - Response: 200
2025-05-28 19:34:47,101 - fire_safety - INFO - Response: 200
2025-05-28 19:34:48,745 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/2
2025-05-28 19:34:48,746 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/2
2025-05-28 19:34:48,746 - fire_safety - INFO - Response: 200
2025-05-28 19:34:48,748 - fire_safety - INFO - Response: 200
2025-05-28 19:34:48,751 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:34:48,761 - fire_safety - INFO - Response: 200
2025-05-28 19:34:48,768 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:34:48,772 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:34:48,774 - fire_safety - INFO - Response: 200
2025-05-28 19:34:48,777 - fire_safety - INFO - Response: 200
2025-05-28 19:34:48,785 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:34:48,786 - fire_safety - INFO - Response: 307
2025-05-28 19:34:48,792 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:34:48,794 - fire_safety - INFO - Response: 307
2025-05-28 19:34:48,796 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:34:48,796 - fire_safety - INFO - Response: 200
2025-05-28 19:34:48,798 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:34:48,805 - fire_safety - INFO - Response: 200
2025-05-28 19:34:48,807 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:34:48,813 - fire_safety - INFO - Response: 200
2025-05-28 19:34:54,183 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:54,189 - fire_safety - INFO - Response: 200
2025-05-28 19:34:54,198 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:54,199 - fire_safety - INFO - Response: 307
2025-05-28 19:34:54,201 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:54,211 - fire_safety - INFO - Response: 200
2025-05-28 19:34:54,215 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:54,216 - fire_safety - INFO - Response: 307
2025-05-28 19:34:54,218 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:54,228 - fire_safety - INFO - Response: 200
2025-05-28 19:34:54,230 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:54,239 - fire_safety - INFO - Response: 200
2025-05-28 19:34:55,203 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:34:55,215 - fire_safety - INFO - Response: 200
2025-05-28 19:34:55,217 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:34:55,224 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:34:55,225 - fire_safety - INFO - Response: 307
2025-05-28 19:34:55,230 - fire_safety - INFO - Response: 200
2025-05-28 19:34:55,233 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:34:55,240 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:34:55,241 - fire_safety - INFO - Response: 307
2025-05-28 19:34:55,243 - fire_safety - INFO - Response: 200
2025-05-28 19:34:55,251 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:34:55,256 - fire_safety - INFO - Response: 200
2025-05-28 19:34:56,080 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:56,081 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:56,082 - fire_safety - INFO - Response: 307
2025-05-28 19:34:56,084 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:56,086 - fire_safety - INFO - Response: 307
2025-05-28 19:34:56,094 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:56,099 - fire_safety - INFO - Response: 200
2025-05-28 19:34:56,106 - fire_safety - INFO - Response: 200
2025-05-28 19:34:56,107 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:56,108 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:56,116 - fire_safety - INFO - Response: 200
2025-05-28 19:34:56,119 - fire_safety - INFO - Response: 200
2025-05-28 19:34:56,808 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:34:56,815 - fire_safety - INFO - Response: 200
2025-05-28 19:34:56,820 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:34:56,824 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:34:56,825 - fire_safety - INFO - Response: 307
2025-05-28 19:34:56,829 - fire_safety - INFO - Response: 200
2025-05-28 19:34:56,832 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:34:56,840 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:34:56,841 - fire_safety - INFO - Response: 307
2025-05-28 19:34:56,842 - fire_safety - INFO - Response: 200
2025-05-28 19:34:56,853 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:34:56,860 - fire_safety - INFO - Response: 200
2025-05-28 19:34:59,957 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:59,959 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:59,961 - fire_safety - INFO - Response: 307
2025-05-28 19:34:59,964 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:34:59,965 - fire_safety - INFO - Response: 307
2025-05-28 19:34:59,970 - fire_safety - INFO - Response: 200
2025-05-28 19:34:59,975 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:34:59,976 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:34:59,990 - fire_safety - INFO - Response: 200
2025-05-28 19:34:59,991 - fire_safety - INFO - Response: 200
2025-05-28 19:34:59,994 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:00,003 - fire_safety - INFO - Response: 200
2025-05-28 19:35:01,313 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:01,325 - fire_safety - INFO - Response: 200
2025-05-28 19:35:01,327 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:01,336 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:01,337 - fire_safety - INFO - Response: 307
2025-05-28 19:35:01,338 - fire_safety - INFO - Response: 200
2025-05-28 19:35:01,348 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:01,349 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:01,350 - fire_safety - INFO - Response: 307
2025-05-28 19:35:01,357 - fire_safety - INFO - Response: 200
2025-05-28 19:35:01,361 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:01,366 - fire_safety - INFO - Response: 200
2025-05-28 19:35:02,492 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:02,497 - fire_safety - INFO - Response: 200
2025-05-28 19:35:02,503 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:02,510 - fire_safety - INFO - Response: 200
2025-05-28 19:35:06,981 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:06,992 - fire_safety - INFO - Response: 200
2025-05-28 19:35:06,996 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:07,008 - fire_safety - INFO - Response: 200
2025-05-28 19:35:07,009 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:07,010 - fire_safety - INFO - Response: 307
2025-05-28 19:35:07,016 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:07,017 - fire_safety - INFO - Response: 307
2025-05-28 19:35:07,020 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:07,027 - fire_safety - INFO - Response: 200
2025-05-28 19:35:07,029 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:07,034 - fire_safety - INFO - Response: 200
2025-05-28 19:35:11,021 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:11,030 - fire_safety - INFO - Response: 200
2025-05-28 19:35:11,032 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:11,038 - fire_safety - INFO - Response: 200
2025-05-28 19:35:13,432 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/1
2025-05-28 19:35:13,432 - fire_safety - INFO - Response: 200
2025-05-28 19:35:13,435 - fire_safety - INFO - Request: PUT http://localhost:8000/sensors/1
2025-05-28 19:35:13,450 - fire_safety - INFO - Response: 200
2025-05-28 19:35:13,472 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:13,478 - fire_safety - INFO - Response: 200
2025-05-28 19:35:13,486 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:13,491 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:13,492 - fire_safety - INFO - Response: 307
2025-05-28 19:35:13,495 - fire_safety - INFO - Response: 200
2025-05-28 19:35:13,503 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:13,506 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:13,511 - fire_safety - INFO - Response: 307
2025-05-28 19:35:13,513 - fire_safety - INFO - Response: 200
2025-05-28 19:35:13,517 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:13,526 - fire_safety - INFO - Response: 200
2025-05-28 19:35:16,135 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:16,138 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:35:16,150 - fire_safety - INFO - Response: 422
2025-05-28 19:35:16,151 - fire_safety - INFO - Response: 200
2025-05-28 19:35:16,154 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:35:16,155 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:16,157 - fire_safety - INFO - Response: 422
2025-05-28 19:35:16,168 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:16,169 - fire_safety - INFO - Response: 307
2025-05-28 19:35:16,171 - fire_safety - INFO - Response: 200
2025-05-28 19:35:16,177 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:16,181 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:16,182 - fire_safety - INFO - Response: 307
2025-05-28 19:35:16,183 - fire_safety - INFO - Response: 200
2025-05-28 19:35:16,190 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:16,195 - fire_safety - INFO - Response: 200
2025-05-28 19:35:17,013 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:17,017 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:35:17,020 - fire_safety - INFO - Response: 422
2025-05-28 19:35:17,027 - fire_safety - INFO - Response: 200
2025-05-28 19:35:17,034 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:35:17,036 - fire_safety - INFO - Response: 422
2025-05-28 19:35:17,037 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:17,046 - fire_safety - INFO - Response: 200
2025-05-28 19:35:17,047 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:17,048 - fire_safety - INFO - Response: 307
2025-05-28 19:35:17,054 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:17,056 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:17,057 - fire_safety - INFO - Response: 307
2025-05-28 19:35:17,062 - fire_safety - INFO - Response: 200
2025-05-28 19:35:17,064 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:17,070 - fire_safety - INFO - Response: 200
2025-05-28 19:35:22,970 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:22,976 - fire_safety - INFO - Response: 200
2025-05-28 19:35:22,980 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:22,985 - fire_safety - INFO - Response: 200
2025-05-28 19:35:28,381 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:28,389 - fire_safety - INFO - Response: 200
2025-05-28 19:35:28,391 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/1
2025-05-28 19:35:28,400 - fire_safety - INFO - Response: 200
2025-05-28 19:35:28,402 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:28,403 - fire_safety - INFO - Response: 307
2025-05-28 19:35:28,412 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=1
2025-05-28 19:35:28,413 - fire_safety - INFO - Response: 307
2025-05-28 19:35:28,414 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:28,420 - fire_safety - INFO - Response: 200
2025-05-28 19:35:28,422 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=1
2025-05-28 19:35:28,427 - fire_safety - INFO - Response: 200
2025-05-28 19:35:32,386 - fire_safety - INFO - Request: DELETE http://localhost:8000/sensors/1
2025-05-28 19:35:32,742 - fire_safety - INFO - Response: 204
2025-05-28 19:35:32,766 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:35:32,768 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:35:32,770 - fire_safety - INFO - Response: 307
2025-05-28 19:35:32,772 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:35:32,774 - fire_safety - INFO - Response: 307
2025-05-28 19:35:32,776 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:32,785 - fire_safety - INFO - Response: 200
2025-05-28 19:35:32,787 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:35:32,790 - fire_safety - INFO - Response: 200
2025-05-28 19:35:32,792 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:32,798 - fire_safety - INFO - Response: 200
2025-05-28 19:35:32,801 - fire_safety - INFO - Response: 200
2025-05-28 19:35:36,507 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:35:36,510 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:35:36,511 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:35:36,512 - fire_safety - INFO - Response: 307
2025-05-28 19:35:36,514 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:35:36,517 - fire_safety - INFO - Response: 307
2025-05-28 19:35:36,519 - fire_safety - INFO - Response: 422
2025-05-28 19:35:36,521 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:36,522 - fire_safety - INFO - Response: 200
2025-05-28 19:35:36,523 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:35:36,525 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:35:36,527 - fire_safety - INFO - Response: 422
2025-05-28 19:35:36,534 - fire_safety - INFO - Response: 200
2025-05-28 19:35:36,536 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:36,537 - fire_safety - INFO - Response: 200
2025-05-28 19:35:36,544 - fire_safety - INFO - Response: 200
2025-05-28 19:35:41,864 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:35:41,865 - fire_safety - INFO - Response: 307
2025-05-28 19:35:41,869 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:35:41,870 - fire_safety - INFO - Response: 307
2025-05-28 19:35:41,876 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:35:41,885 - fire_safety - INFO - Response: 200
2025-05-28 19:35:41,887 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:35:41,893 - fire_safety - INFO - Response: 200
2025-05-28 19:35:45,542 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:35:45,552 - fire_safety - INFO - Response: 200
2025-05-28 19:35:45,555 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:35:45,557 - fire_safety - INFO - Response: 307
2025-05-28 19:35:45,558 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:35:45,566 - fire_safety - INFO - Response: 200
2025-05-28 19:35:45,568 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:35:45,569 - fire_safety - INFO - Response: 307
2025-05-28 19:35:45,571 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:45,580 - fire_safety - INFO - Response: 200
2025-05-28 19:35:45,581 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:45,590 - fire_safety - INFO - Response: 200
2025-05-28 19:35:50,994 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:35:50,995 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:35:50,997 - fire_safety - INFO - Response: 307
2025-05-28 19:35:51,003 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:35:51,005 - fire_safety - INFO - Response: 307
2025-05-28 19:35:51,007 - fire_safety - INFO - Response: 200
2025-05-28 19:35:51,009 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:51,010 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:35:51,025 - fire_safety - INFO - Response: 200
2025-05-28 19:35:51,026 - fire_safety - INFO - Response: 200
2025-05-28 19:35:51,030 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:35:51,035 - fire_safety - INFO - Response: 200
2025-05-28 19:35:55,473 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:35:55,480 - fire_safety - INFO - Response: 200
2025-05-28 19:35:55,486 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:35:55,491 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:35:55,492 - fire_safety - INFO - Response: 307
2025-05-28 19:35:55,494 - fire_safety - INFO - Response: 200
2025-05-28 19:35:55,501 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:35:55,504 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:35:55,505 - fire_safety - INFO - Response: 307
2025-05-28 19:35:55,508 - fire_safety - INFO - Response: 200
2025-05-28 19:35:55,512 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:35:55,518 - fire_safety - INFO - Response: 200
2025-05-28 19:57:17,489 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/2
2025-05-28 19:57:17,490 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:57:17,491 - fire_safety - INFO - Response: 200
2025-05-28 19:57:17,492 - fire_safety - INFO - Response: 200
2025-05-28 19:57:17,495 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:57:17,500 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:57:17,501 - fire_safety - INFO - Response: 307
2025-05-28 19:57:17,504 - fire_safety - INFO - Response: 200
2025-05-28 19:57:17,509 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:57:17,510 - fire_safety - INFO - Response: 200
2025-05-28 19:57:17,513 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:57:17,518 - fire_safety - INFO - Response: 200
2025-05-28 19:57:17,527 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 19:57:17,527 - fire_safety - INFO - Response: 200
2025-05-28 19:57:17,529 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:57:17,537 - fire_safety - INFO - Response: 200
2025-05-28 19:57:37,926 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/token
2025-05-28 19:57:37,927 - fire_safety - INFO - Response: 200
2025-05-28 19:57:37,930 - fire_safety - INFO - Request: POST http://localhost:8000/token
2025-05-28 19:57:37,940 - fire_safety - INFO - Response: 422
2025-05-28 19:57:45,767 - fire_safety - INFO - Request: POST http://localhost:8000/token
2025-05-28 19:57:46,177 - fire_safety - INFO - Response: 200
2025-05-28 19:57:47,594 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 19:57:47,595 - fire_safety - INFO - Response: 200
2025-05-28 19:57:47,597 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 19:57:47,598 - fire_safety - INFO - Response: 200
2025-05-28 19:57:47,600 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:57:47,601 - fire_safety - INFO - Response: 307
2025-05-28 19:57:47,608 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 19:57:47,609 - fire_safety - INFO - Response: 307
2025-05-28 19:57:47,611 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 19:57:47,612 - fire_safety - INFO - Response: 200
2025-05-28 19:57:47,618 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:57:47,624 - fire_safety - INFO - Response: 200
2025-05-28 19:57:47,630 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 19:57:47,637 - fire_safety - INFO - Response: 200
2025-05-28 19:57:49,803 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:57:49,805 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 19:57:49,806 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 19:57:49,807 - fire_safety - INFO - Response: 200
2025-05-28 19:57:49,809 - fire_safety - INFO - Response: 200
2025-05-28 19:57:49,811 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:57:49,812 - fire_safety - INFO - Response: 307
2025-05-28 19:57:49,821 - fire_safety - INFO - Response: 200
2025-05-28 19:57:49,823 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:57:49,824 - fire_safety - INFO - Response: 307
2025-05-28 19:57:49,825 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:57:49,827 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 19:57:49,828 - fire_safety - INFO - Response: 200
2025-05-28 19:57:49,832 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 19:57:49,833 - fire_safety - INFO - Response: 200
2025-05-28 19:57:49,834 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:57:49,846 - fire_safety - INFO - Response: 200
2025-05-28 19:57:49,849 - fire_safety - INFO - Response: 200
2025-05-28 19:57:49,851 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:57:49,858 - fire_safety - INFO - Response: 200
2025-05-28 19:57:50,916 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:57:50,918 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:57:50,919 - fire_safety - INFO - Response: 307
2025-05-28 19:57:50,929 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:57:50,930 - fire_safety - INFO - Response: 307
2025-05-28 19:57:50,932 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:57:50,935 - fire_safety - INFO - Response: 200
2025-05-28 19:57:50,942 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:57:50,944 - fire_safety - INFO - Response: 200
2025-05-28 19:57:50,946 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:57:50,956 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:57:50,964 - fire_safety - INFO - Response: 200
2025-05-28 19:57:50,968 - fire_safety - INFO - Response: 200
2025-05-28 19:57:50,971 - fire_safety - INFO - Response: 200
2025-05-28 19:57:50,978 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:57:50,984 - fire_safety - INFO - Response: 200
2025-05-28 19:57:55,616 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:57:55,617 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:57:55,620 - fire_safety - INFO - Response: 307
2025-05-28 19:57:55,625 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 19:57:55,629 - fire_safety - INFO - Response: 307
2025-05-28 19:57:55,636 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:57:55,638 - fire_safety - INFO - Response: 200
2025-05-28 19:57:55,639 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:57:55,648 - fire_safety - INFO - Response: 200
2025-05-28 19:57:55,651 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 19:57:55,654 - fire_safety - INFO - Response: 200
2025-05-28 19:57:55,659 - fire_safety - INFO - Response: 200
2025-05-28 19:58:00,270 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:58:00,276 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:58:00,279 - fire_safety - INFO - Response: 307
2025-05-28 19:58:00,285 - fire_safety - INFO - Response: 200
2025-05-28 19:58:00,291 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:58:00,292 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:58:00,292 - fire_safety - INFO - Response: 307
2025-05-28 19:58:00,294 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:58:00,303 - fire_safety - INFO - Response: 200
2025-05-28 19:58:00,305 - fire_safety - INFO - Response: 200
2025-05-28 19:58:00,306 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:58:00,315 - fire_safety - INFO - Response: 200
2025-05-28 19:58:00,317 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:58:00,322 - fire_safety - INFO - Response: 200
2025-05-28 19:58:00,324 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:58:00,332 - fire_safety - INFO - Response: 200
2025-05-28 19:58:02,013 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:58:02,015 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 19:58:02,016 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 19:58:02,016 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:58:02,017 - fire_safety - INFO - Response: 200
2025-05-28 19:58:02,019 - fire_safety - INFO - Response: 200
2025-05-28 19:58:02,020 - fire_safety - INFO - Response: 307
2025-05-28 19:58:02,022 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:58:02,025 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 19:58:02,026 - fire_safety - INFO - Response: 307
2025-05-28 19:58:02,030 - fire_safety - INFO - Response: 422
2025-05-28 19:58:02,037 - fire_safety - INFO - Response: 200
2025-05-28 19:58:02,040 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:58:02,041 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 19:58:02,043 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 19:58:02,048 - fire_safety - INFO - Response: 422
2025-05-28 19:58:02,054 - fire_safety - INFO - Response: 200
2025-05-28 19:58:02,056 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 19:58:02,057 - fire_safety - INFO - Response: 200
2025-05-28 19:58:02,064 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:58:02,067 - fire_safety - INFO - Response: 200
2025-05-28 19:58:02,070 - fire_safety - INFO - Response: 200
2025-05-28 19:58:02,078 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 19:58:02,083 - fire_safety - INFO - Response: 200
2025-05-28 20:00:13,354 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:13,355 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:13,356 - fire_safety - INFO - Response: 307
2025-05-28 20:00:13,359 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:13,360 - fire_safety - INFO - Response: 307
2025-05-28 20:00:13,363 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:13,373 - fire_safety - INFO - Response: 200
2025-05-28 20:00:13,376 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:13,378 - fire_safety - INFO - Response: 200
2025-05-28 20:00:13,382 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:13,387 - fire_safety - INFO - Response: 200
2025-05-28 20:00:13,390 - fire_safety - INFO - Response: 200
2025-05-28 20:00:16,732 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:16,733 - fire_safety - INFO - Response: 307
2025-05-28 20:00:16,735 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:16,736 - fire_safety - INFO - Response: 307
2025-05-28 20:00:16,761 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:16,772 - fire_safety - INFO - Response: 200
2025-05-28 20:00:16,811 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:16,817 - fire_safety - INFO - Response: 200
2025-05-28 20:00:20,755 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:20,757 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:20,759 - fire_safety - INFO - Response: 307
2025-05-28 20:00:20,763 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:20,764 - fire_safety - INFO - Response: 307
2025-05-28 20:00:20,767 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:20,774 - fire_safety - INFO - Response: 200
2025-05-28 20:00:20,779 - fire_safety - INFO - Response: 200
2025-05-28 20:00:20,781 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:20,786 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:20,790 - fire_safety - INFO - Response: 200
2025-05-28 20:00:20,793 - fire_safety - INFO - Response: 200
2025-05-28 20:00:30,552 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/
2025-05-28 20:00:30,553 - fire_safety - INFO - Response: 200
2025-05-28 20:00:30,555 - fire_safety - INFO - Request: POST http://localhost:8000/sensors/
2025-05-28 20:00:30,715 - fire_safety - INFO - Response: 201
2025-05-28 20:00:30,741 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:30,749 - fire_safety - INFO - Response: 200
2025-05-28 20:00:30,755 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:30,755 - fire_safety - INFO - Response: 307
2025-05-28 20:00:30,757 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:30,765 - fire_safety - INFO - Response: 200
2025-05-28 20:00:30,768 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:30,769 - fire_safety - INFO - Response: 307
2025-05-28 20:00:30,771 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:30,778 - fire_safety - INFO - Response: 200
2025-05-28 20:00:30,783 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:30,789 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,347 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/3
2025-05-28 20:00:32,348 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,349 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=3
2025-05-28 20:00:32,351 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/3
2025-05-28 20:00:32,351 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=3
2025-05-28 20:00:32,352 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,353 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/3
2025-05-28 20:00:32,354 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,354 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,356 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=3
2025-05-28 20:00:32,359 - fire_safety - INFO - Response: 307
2025-05-28 20:00:32,367 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,370 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=3
2025-05-28 20:00:32,371 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/3
2025-05-28 20:00:32,372 - fire_safety - INFO - Response: 307
2025-05-28 20:00:32,375 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents/?sensor_id=3
2025-05-28 20:00:32,376 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,379 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=3
2025-05-28 20:00:32,395 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,396 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,397 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=3
2025-05-28 20:00:32,407 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:32,410 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,415 - fire_safety - INFO - Response: 200
2025-05-28 20:00:32,420 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:32,426 - fire_safety - INFO - Response: 200
2025-05-28 20:00:41,830 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents
2025-05-28 20:00:41,831 - fire_safety - INFO - Response: 200
2025-05-28 20:00:41,834 - fire_safety - INFO - Request: POST http://localhost:8000/incidents
2025-05-28 20:00:41,835 - fire_safety - INFO - Response: 307
2025-05-28 20:00:41,840 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents/
2025-05-28 20:00:41,841 - fire_safety - INFO - Response: 200
2025-05-28 20:00:41,843 - fire_safety - INFO - Request: POST http://localhost:8000/incidents/
2025-05-28 20:00:41,879 - fire_safety - ERROR - Request failed: 1 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 1 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:00:41,909 - fire_safety - ERROR - Unhandled exception: 1 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:00:43,439 - fire_safety - INFO - Request: POST http://localhost:8000/incidents
2025-05-28 20:00:43,439 - fire_safety - INFO - Response: 307
2025-05-28 20:00:43,445 - fire_safety - INFO - Request: POST http://localhost:8000/incidents/
2025-05-28 20:00:43,454 - fire_safety - ERROR - Request failed: 1 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 1 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:00:43,462 - fire_safety - ERROR - Unhandled exception: 1 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:00:45,467 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:45,468 - fire_safety - INFO - Response: 307
2025-05-28 20:00:45,469 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:45,470 - fire_safety - INFO - Response: 307
2025-05-28 20:00:45,483 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:45,495 - fire_safety - INFO - Response: 200
2025-05-28 20:00:45,504 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:45,509 - fire_safety - INFO - Response: 200
2025-05-28 20:00:47,395 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:47,398 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:47,400 - fire_safety - INFO - Response: 307
2025-05-28 20:00:47,404 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:47,407 - fire_safety - INFO - Response: 307
2025-05-28 20:00:47,415 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:47,416 - fire_safety - INFO - Response: 200
2025-05-28 20:00:47,423 - fire_safety - INFO - Response: 200
2025-05-28 20:00:47,426 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:47,427 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:47,440 - fire_safety - INFO - Response: 200
2025-05-28 20:00:47,441 - fire_safety - INFO - Response: 200
2025-05-28 20:00:49,244 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 20:00:49,247 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 20:00:49,248 - fire_safety - INFO - Response: 307
2025-05-28 20:00:49,251 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 20:00:49,253 - fire_safety - INFO - Response: 307
2025-05-28 20:00:49,263 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 20:00:49,267 - fire_safety - INFO - Response: 200
2025-05-28 20:00:49,269 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 20:00:49,270 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:00:49,279 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:00:49,300 - fire_safety - INFO - Response: 200
2025-05-28 20:00:49,303 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 20:00:49,308 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:00:49,314 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:00:51,099 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:51,101 - fire_safety - INFO - Response: 307
2025-05-28 20:00:51,108 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:51,109 - fire_safety - INFO - Response: 307
2025-05-28 20:00:51,112 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:51,122 - fire_safety - INFO - Response: 200
2025-05-28 20:00:51,133 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:51,140 - fire_safety - INFO - Response: 200
2025-05-28 20:00:52,160 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:52,161 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:52,162 - fire_safety - INFO - Response: 307
2025-05-28 20:00:52,164 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:52,166 - fire_safety - INFO - Response: 307
2025-05-28 20:00:52,174 - fire_safety - INFO - Response: 200
2025-05-28 20:00:52,177 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:52,178 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:52,186 - fire_safety - INFO - Response: 200
2025-05-28 20:00:52,189 - fire_safety - INFO - Response: 200
2025-05-28 20:00:52,191 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:52,197 - fire_safety - INFO - Response: 200
2025-05-28 20:00:52,850 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/3
2025-05-28 20:00:52,854 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=3
2025-05-28 20:00:52,856 - fire_safety - INFO - Response: 307
2025-05-28 20:00:52,858 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=3
2025-05-28 20:00:52,861 - fire_safety - INFO - Response: 307
2025-05-28 20:00:52,867 - fire_safety - INFO - Response: 200
2025-05-28 20:00:52,868 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=3
2025-05-28 20:00:52,874 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/3
2025-05-28 20:00:52,876 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:00:52,882 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:00:52,902 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=3
2025-05-28 20:00:52,904 - fire_safety - INFO - Response: 200
2025-05-28 20:00:52,908 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:00:52,915 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:00:53,899 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:53,901 - fire_safety - INFO - Response: 307
2025-05-28 20:00:53,905 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:53,906 - fire_safety - INFO - Response: 307
2025-05-28 20:00:53,912 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:53,920 - fire_safety - INFO - Response: 200
2025-05-28 20:00:53,922 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:53,927 - fire_safety - INFO - Response: 200
2025-05-28 20:00:54,912 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:54,913 - fire_safety - INFO - Response: 307
2025-05-28 20:00:54,917 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 20:00:54,920 - fire_safety - INFO - Response: 422
2025-05-28 20:00:54,923 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:00:54,924 - fire_safety - INFO - Response: 307
2025-05-28 20:00:54,926 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 20:00:54,929 - fire_safety - INFO - Response: 422
2025-05-28 20:00:54,933 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:54,937 - fire_safety - INFO - Response: 200
2025-05-28 20:00:54,939 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:00:54,946 - fire_safety - INFO - Response: 200
2025-05-28 20:00:57,748 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:57,753 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:57,754 - fire_safety - INFO - Response: 307
2025-05-28 20:00:57,760 - fire_safety - INFO - Response: 200
2025-05-28 20:00:57,762 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:00:57,763 - fire_safety - INFO - Response: 307
2025-05-28 20:00:57,767 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:57,774 - fire_safety - INFO - Response: 200
2025-05-28 20:00:57,776 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:00:57,783 - fire_safety - INFO - Response: 200
2025-05-28 20:00:57,786 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:00:57,792 - fire_safety - INFO - Response: 200
2025-05-28 20:00:58,439 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 20:00:58,443 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 20:00:58,444 - fire_safety - INFO - Response: 307
2025-05-28 20:00:58,446 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 20:00:58,447 - fire_safety - INFO - Response: 307
2025-05-28 20:00:58,448 - fire_safety - INFO - Response: 200
2025-05-28 20:00:58,454 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 20:00:58,455 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 20:00:58,462 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:00:58,469 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:00:58,469 - fire_safety - INFO - Response: 200
2025-05-28 20:00:58,487 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 20:00:58,493 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:00:58,498 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:01:06,413 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 20:01:06,423 - fire_safety - INFO - Response: 200
2025-05-28 20:01:06,427 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 20:01:06,430 - fire_safety - INFO - Response: 307
2025-05-28 20:01:06,433 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 20:01:06,435 - fire_safety - INFO - Response: 422
2025-05-28 20:01:06,438 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 20:01:06,446 - fire_safety - INFO - Response: 200
2025-05-28 20:01:06,447 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 20:01:06,448 - fire_safety - INFO - Response: 307
2025-05-28 20:01:06,450 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 20:01:06,456 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:01:06,461 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:01:06,724 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 20:01:06,726 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 20:01:06,738 - fire_safety - INFO - Response: 422
2025-05-28 20:01:06,742 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:01:06,749 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:01:13,624 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:01:13,625 - fire_safety - INFO - Response: 307
2025-05-28 20:01:13,628 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 20:01:13,630 - fire_safety - INFO - Response: 307
2025-05-28 20:01:13,639 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:01:13,671 - fire_safety - INFO - Response: 200
2025-05-28 20:01:13,673 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 20:01:13,679 - fire_safety - INFO - Response: 200
2025-05-28 20:01:14,664 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:01:14,665 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:01:14,666 - fire_safety - INFO - Response: 307
2025-05-28 20:01:14,670 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:01:14,677 - fire_safety - INFO - Response: 307
2025-05-28 20:01:14,682 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:01:14,691 - fire_safety - INFO - Response: 200
2025-05-28 20:01:14,699 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:01:14,706 - fire_safety - INFO - Response: 200
2025-05-28 20:01:14,716 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:01:14,721 - fire_safety - INFO - Response: 200
2025-05-28 20:01:14,781 - fire_safety - INFO - Response: 200
2025-05-28 20:01:16,135 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/3
2025-05-28 20:01:16,138 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=3
2025-05-28 20:01:16,140 - fire_safety - INFO - Response: 307
2025-05-28 20:01:16,143 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=3
2025-05-28 20:01:16,145 - fire_safety - INFO - Response: 307
2025-05-28 20:01:16,153 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=3
2025-05-28 20:01:16,156 - fire_safety - INFO - Response: 200
2025-05-28 20:01:16,161 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/3
2025-05-28 20:01:16,166 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:01:16,175 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:01:16,199 - fire_safety - INFO - Response: 200
2025-05-28 20:01:16,200 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=3
2025-05-28 20:01:16,207 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 318, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 20:01:16,213 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 20:01:17,581 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:01:17,590 - fire_safety - INFO - Response: 200
2025-05-28 20:01:17,603 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:01:17,604 - fire_safety - INFO - Response: 307
2025-05-28 20:01:17,606 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 20:01:17,612 - fire_safety - INFO - Response: 200
2025-05-28 20:01:17,614 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 20:01:17,615 - fire_safety - INFO - Response: 307
2025-05-28 20:01:17,616 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:01:17,624 - fire_safety - INFO - Response: 200
2025-05-28 20:01:17,626 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 20:01:17,633 - fire_safety - INFO - Response: 200
2025-05-28 23:21:21,994 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 23:21:21,995 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 23:21:21,995 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 23:21:21,996 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 23:21:21,996 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 23:21:21,997 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 23:21:21,997 - fire_safety - INFO - Response: 200
2025-05-28 23:21:21,998 - fire_safety - INFO - Response: 200
2025-05-28 23:21:21,999 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,000 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,001 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,002 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,005 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 23:21:22,005 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 23:21:22,006 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,007 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,008 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 23:21:22,008 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 23:21:22,009 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 23:21:22,009 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 23:21:22,010 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 23:21:22,011 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,011 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,012 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 23:21:22,013 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,014 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,015 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,016 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,017 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 23:21:22,017 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 23:21:22,019 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 23:21:22,020 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:21:22,026 - fire_safety - INFO - Response: 307
2025-05-28 23:21:22,028 - fire_safety - INFO - Response: 307
2025-05-28 23:21:22,029 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 23:21:22,049 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:21:22,049 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 23:21:22,050 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 23:21:22,050 - fire_safety - INFO - Response: 307
2025-05-28 23:21:22,052 - fire_safety - INFO - Response: 422
2025-05-28 23:21:22,053 - fire_safety - INFO - Response: 307
2025-05-28 23:21:22,054 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,054 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,056 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 23:21:22,058 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:21:22,061 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 23:21:22,062 - fire_safety - INFO - Response: 422
2025-05-28 23:21:22,063 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,068 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 23:21:22,072 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,078 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:21:22,084 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,143 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,144 - fire_safety - INFO - Response: 200
2025-05-28 23:21:22,145 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 23:21:22,151 - fire_safety - INFO - Response: 200
2025-05-28 23:21:28,450 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:21:28,451 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 23:21:28,451 - fire_safety - INFO - Response: 307
2025-05-28 23:21:28,454 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:21:28,454 - fire_safety - INFO - Response: 422
2025-05-28 23:21:28,456 - fire_safety - INFO - Response: 307
2025-05-28 23:21:28,460 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 23:21:28,463 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:21:28,467 - fire_safety - INFO - Response: 422
2025-05-28 23:21:28,472 - fire_safety - INFO - Response: 200
2025-05-28 23:21:28,477 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:21:28,483 - fire_safety - INFO - Response: 200
2025-05-28 23:36:59,053 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 23:36:59,054 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/users/me
2025-05-28 23:36:59,054 - fire_safety - INFO - Response: 200
2025-05-28 23:36:59,055 - fire_safety - INFO - Response: 200
2025-05-28 23:36:59,057 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 23:36:59,066 - fire_safety - INFO - Response: 422
2025-05-28 23:36:59,068 - fire_safety - INFO - Request: GET http://localhost:8000/users/me
2025-05-28 23:36:59,070 - fire_safety - INFO - Response: 422
2025-05-28 23:37:13,604 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 23:37:13,605 - fire_safety - INFO - Response: 200
2025-05-28 23:37:13,607 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings
2025-05-28 23:37:13,608 - fire_safety - INFO - Response: 200
2025-05-28 23:37:13,610 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:37:13,612 - fire_safety - INFO - Response: 307
2025-05-28 23:37:13,614 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:37:13,615 - fire_safety - INFO - Response: 307
2025-05-28 23:37:13,620 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/
2025-05-28 23:37:13,621 - fire_safety - INFO - Response: 200
2025-05-28 23:37:13,623 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:37:13,647 - fire_safety - INFO - Response: 200
2025-05-28 23:37:13,649 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:37:13,654 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,216 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 23:37:15,218 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 23:37:15,219 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/buildings/7
2025-05-28 23:37:15,219 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,220 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors?building_id=7
2025-05-28 23:37:15,221 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,222 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,223 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,232 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 23:37:15,233 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 23:37:15,234 - fire_safety - INFO - Response: 307
2025-05-28 23:37:15,236 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 23:37:15,238 - fire_safety - INFO - Response: 307
2025-05-28 23:37:15,242 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 23:37:15,245 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,246 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,248 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/?building_id=7
2025-05-28 23:37:15,248 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,251 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 23:37:15,252 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 23:37:15,267 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,274 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 23:37:15,279 - fire_safety - INFO - Response: 200
2025-05-28 23:37:15,328 - fire_safety - INFO - Response: 200
2025-05-28 23:37:16,573 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/2
2025-05-28 23:37:16,574 - fire_safety - INFO - Response: 200
2025-05-28 23:37:16,575 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=2
2025-05-28 23:37:16,576 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=2
2025-05-28 23:37:16,577 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/2
2025-05-28 23:37:16,578 - fire_safety - INFO - Response: 200
2025-05-28 23:37:16,580 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 23:37:16,580 - fire_safety - INFO - Response: 200
2025-05-28 23:37:16,582 - fire_safety - INFO - Response: 200
2025-05-28 23:37:16,584 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 23:37:16,585 - fire_safety - INFO - Response: 307
2025-05-28 23:37:16,587 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 23:37:16,589 - fire_safety - INFO - Response: 307
2025-05-28 23:37:16,593 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents/?sensor_id=2
2025-05-28 23:37:16,593 - fire_safety - INFO - Response: 200
2025-05-28 23:37:16,596 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 23:37:16,597 - fire_safety - INFO - Response: 200
2025-05-28 23:37:16,599 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 23:37:16,609 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 353, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 23:37:16,635 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 23:37:16,672 - fire_safety - INFO - Response: 200
2025-05-28 23:37:16,674 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 23:37:16,679 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 353, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 23:37:16,684 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 23:37:18,898 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:37:18,899 - fire_safety - INFO - Response: 307
2025-05-28 23:37:18,902 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:37:18,902 - fire_safety - INFO - Response: 307
2025-05-28 23:37:18,908 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:37:18,917 - fire_safety - INFO - Response: 200
2025-05-28 23:37:18,919 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:37:18,927 - fire_safety - INFO - Response: 200
2025-05-28 23:37:20,232 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 23:37:20,234 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 23:37:20,235 - fire_safety - INFO - Response: 307
2025-05-28 23:37:20,239 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 23:37:20,241 - fire_safety - INFO - Response: 307
2025-05-28 23:37:20,245 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 23:37:20,251 - fire_safety - INFO - Response: 200
2025-05-28 23:37:20,254 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 23:37:20,255 - fire_safety - INFO - Response: 200
2025-05-28 23:37:20,258 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 23:37:20,266 - fire_safety - INFO - Response: 200
2025-05-28 23:37:20,269 - fire_safety - INFO - Response: 200
2025-05-28 23:37:20,997 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/3
2025-05-28 23:37:20,998 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=3
2025-05-28 23:37:20,998 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/sensors/3
2025-05-28 23:37:20,999 - fire_safety - INFO - Response: 200
2025-05-28 23:37:21,000 - fire_safety - INFO - Response: 200
2025-05-28 23:37:21,001 - fire_safety - INFO - Response: 200
2025-05-28 23:37:21,002 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents?sensor_id=3
2025-05-28 23:37:21,003 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/3
2025-05-28 23:37:21,004 - fire_safety - INFO - Response: 200
2025-05-28 23:37:21,006 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=3
2025-05-28 23:37:21,008 - fire_safety - INFO - Response: 307
2025-05-28 23:37:21,010 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=3
2025-05-28 23:37:21,011 - fire_safety - INFO - Response: 307
2025-05-28 23:37:21,013 - fire_safety - INFO - Request: OPTIONS http://localhost:8000/incidents/?sensor_id=3
2025-05-28 23:37:21,014 - fire_safety - INFO - Response: 200
2025-05-28 23:37:21,016 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=3
2025-05-28 23:37:21,018 - fire_safety - INFO - Response: 200
2025-05-28 23:37:21,022 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/3
2025-05-28 23:37:21,027 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 353, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 23:37:21,036 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 23:37:21,053 - fire_safety - INFO - Response: 200
2025-05-28 23:37:21,054 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=3
2025-05-28 23:37:21,060 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 353, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 23:37:21,065 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 23:37:22,120 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:37:22,121 - fire_safety - INFO - Response: 307
2025-05-28 23:37:22,122 - fire_safety - INFO - Request: GET http://localhost:8000/buildings
2025-05-28 23:37:22,123 - fire_safety - INFO - Response: 307
2025-05-28 23:37:22,126 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:37:22,134 - fire_safety - INFO - Response: 200
2025-05-28 23:37:22,137 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/
2025-05-28 23:37:22,144 - fire_safety - INFO - Response: 200
2025-05-28 23:37:26,000 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 23:37:26,001 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 23:37:26,002 - fire_safety - INFO - Response: 307
2025-05-28 23:37:26,004 - fire_safety - INFO - Request: GET http://localhost:8000/sensors?building_id=7
2025-05-28 23:37:26,005 - fire_safety - INFO - Response: 307
2025-05-28 23:37:26,011 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 23:37:26,014 - fire_safety - INFO - Response: 200
2025-05-28 23:37:26,017 - fire_safety - INFO - Request: GET http://localhost:8000/buildings/7
2025-05-28 23:37:26,021 - fire_safety - INFO - Response: 200
2025-05-28 23:37:26,024 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/?building_id=7
2025-05-28 23:37:26,027 - fire_safety - INFO - Response: 200
2025-05-28 23:37:26,031 - fire_safety - INFO - Response: 200
2025-05-28 23:37:26,735 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 23:37:26,736 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 23:37:26,737 - fire_safety - INFO - Response: 307
2025-05-28 23:37:26,739 - fire_safety - INFO - Request: GET http://localhost:8000/incidents?sensor_id=2
2025-05-28 23:37:26,742 - fire_safety - INFO - Response: 307
2025-05-28 23:37:26,744 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 23:37:26,747 - fire_safety - INFO - Response: 200
2025-05-28 23:37:26,749 - fire_safety - INFO - Request: GET http://localhost:8000/sensors/2
2025-05-28 23:37:26,753 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 353, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 23:37:26,759 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
2025-05-28 23:37:26,774 - fire_safety - INFO - Response: 200
2025-05-28 23:37:26,776 - fire_safety - INFO - Request: GET http://localhost:8000/incidents/?sensor_id=2
2025-05-28 23:37:26,782 - fire_safety - ERROR - Request failed: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
Traceback (most recent call last):
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 148, in call_next
    message = await recv_stream.receive()
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\anyio\streams\memory.py", line 126, in receive
    raise EndOfStream from None
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Roma\Desktop\À–4 “Ãœ\main.py", line 353, in log_requests
    response = await call_next(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 156, in call_next
    raise app_exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 327, in app
    content = await serialize_response(
  File "C:\Users\Roma\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 176, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}

2025-05-28 23:37:26,787 - fire_safety - ERROR - Unhandled exception: 2 validation errors:
  {'type': 'datetime_type', 'loc': ('response', 0, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
  {'type': 'datetime_type', 'loc': ('response', 1, 'detected_at'), 'msg': 'Input should be a valid datetime', 'input': None}
NoneType: None
